### ISSUE TITLE
Undefined name "Reference" when compiling generated schema

### ISSUE THREAD

**Issue by rubberduck203** [carreragroup/json-to-avro-schema#36]
- Process the fhir json schema
- Try to compile it with `avro-tools`

```console
$ avro-tools compile schema fhir.avsc .
Input files to compile:
  fhir.avsc
Exception in thread "main" org.apache.avro.SchemaParseException: Undefined name: "Reference"
```

I suspect there's something wrong with how we're processing definitions.

It appears that we're not inlining the type as expected in the "Account" type.

```json
{
  "type": "record",
  "name": "FhirResource",
  "doc": "see http://hl7.org/fhir/json.html#schema for information about the FHIR Json Schemas",
  "fields": [
    {
      "name": "value",
      "doc": "see http://hl7.org/fhir/json.html#schema for information about the FHIR Json Schemas",
      "type": [
        {
          "type": "record",
          "name": "Account",
          "doc": "A financial tool for tracking value accrued for a particular purpose.  In the healthcare field, used to track charges for a patient, cost centers, etc.",
          "fields": [
            {
              "name": "name",
              "doc": "Name used for the account when displaying it to humans in reports, etc.",
              "type": [
                "null",
                "string"
              ],
              "default": null
            },
            {
              "name": "subject",
              "doc": "Identifies the entity which incurs the expenses. While the immediate recipients of services or goods might be entities related to the subject, the expenses were ultimately incurred by the subject of the Account.",
              "type": [
                "null",
                {
                  "type": "array",
                  "items": "Reference"
                }
              ],
              "default": null
            },
 ```

There is a `reference` type defined much later (around line 30318), but I'm not confident that's the type that `Account` is refering to.

```json
          "type": "record",
          "name": "DetectedIssue",
          "doc": "Indicates an actual or potential clinical issue with or between one or more active or proposed clinical actions for a patient; e.g. Drug-drug interaction, Ineffective treatment frequency, Procedure-condition conflict, etc.",
          "fields": [
            {
              "name": "reference",
              "doc": "The literature, knowledge-base or similar reference that describes the propensity for the detected issue identified.",
              "type": [
                "null",
                "uri"
              ],
              "default": null
            },
  ```

**Comment by rubberduck203** [comment]
I think we're just not replacing the first instance of a def ref if it's inside an `items`.
This should be a good test case.

```json
{
  "definitions": {
    "Account": {
      "properties": {
        "subject": {
          "description": "Identifies the entity which incurs the expenses. While the immediate recipients of services or goods might be entities related to the subject, the expenses were ultimately incurred by the subject of the Account.",
          "items": {
            "$ref": "#/definitions/Reference"
          },
          "type": "array"
        }
      }
    }
  }
}
```

**Comment by rubberduck203** [comment]
Closed by #37
